addons/mail/models/mail_thread.py

```diff
@@ -1286,6 +1286,11 @@ def _message_parse_extract_payload(self, message, save_original=False):
                     continue  # skip container
 
                 filename = part.get_filename()  # I may not properly handle all charsets
+                ascii = not part.get_param('charset') or part.get_param('charset').lower() in ('us-ascii', 'ascii')
+                if part.get_content_type() == 'text/xml' and ascii and part.get_payload(decode=True).startswith(b"\xef\xbb\xbf"):
+                    # for text/xml with omitted charset, the charset is assumed to be ascii. If the payload contains
+                    # a BOM, it will be replaced by replacement chars, preventing the parsing of the xml attachment.
+                    part.set_charset('utf-8')
```

Comment:
I was wondering why we cared about mrp "Bills of Material" in here but now it makes sense, we don't :D

It turns out every encoding (event utf-8 and utf-16) have their own BOM magic number, and from what I've seen ASCII should not have any. Did we have that case? Otherwise we could just do this when there is no charset.

I would update the comment to just say "If the content starts with the UTF-8 BOM, mark the part as UTF-8. Converting it directly to ASCII would result in invalid characters, leading to a crash."

Reply:
We only encountered the case of an omitted charset and a payload in UTF-8 starting with the UTF-8 BOM. I wanted to be "safe" by also including the case when ASCII is explicitely specified as the charset as well, but I guess it not necessary (at least for now) :)

Reply:
> It turns out every encoding (event utf-8 and utf-16) have their own BOM magic number,

Indeed but must of the time it is omitted in utf-8 so that the string is compatible with ascii when restricting to the US-ASCII script and so compatible with most pre-unicode softwares